### PR TITLE
8338740: java/net/httpclient/HttpsTunnelAuthTest.java fails with java.io.IOException: HTTP/1.1 header parser received no bytes

### PR DESCRIPTION
**Pull request by dfuch**
The `java/net/httpclient/HttpsTunnelAuthTest.java` has been observed failing intermittently with `java.io.IOException: HTTP/1.1 header parser received no bytes`, ... `Caused by: java.net.SocketException: Connection reset `.

My suspicion is that the ProxyServer used by this test is the cause of the reset: when a tunnel connection is established, the proxy server acts as an intermediary between the client and the server, and shuffles things around in two directions between two sockets. However, if one end of one of the two sockets gets closed, the proxy will abruptly close the two sockets, without waiting for inflight traffic on the other direction to quiesce. I believe this is what is causing the connection reset.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8338740](https://bugs.openjdk.org/browse/JDK-8338740): java/net/httpclient/HttpsTunnelAuthTest.java fails with java.io.IOException: HTTP/1.1 header parser received no bytes (**Bug** - P4)

### Reviewers
 * [Daniel Jeliński](https://openjdk.org/census#djelinski) (@djelinski - **Reviewer**)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20674/head:pull/20674` \
`$ git checkout pull/20674`

Update a local copy of the PR: \
`$ git checkout pull/20674` \
`$ git pull https://git.openjdk.org/jdk.git pull/20674/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20674`

View PR using the GUI difftool: \
`$ git pr show -t 20674`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20674.diff">https://git.openjdk.org/jdk/pull/20674.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20674#issuecomment-2315679221)

### CODE REVIEW

File: test/jdk/java/net/httpclient/ProxyServer.java [hunk 1]
```diff
@@ -195,6 +195,9 @@ class Connection {
         volatile InputStream clientIn, serverIn;
         volatile OutputStream clientOut, serverOut;
 
+        volatile boolean proxyInClosed;
+        volatile boolean proxyOutClosed;
```
Comment (djelinski) [inline test/jdk/java/net/httpclient/ProxyServer.java:199]: nit: these variables don't need to be volatile; they are only accessed in a synchronized block

**Review by djelinski — APPROVED** [review]
LGTM. Please update the copyright before integrating.

File: test/jdk/java/net/httpclient/ProxyServer.java (hunk 1 above)
Comment (dfuch) [reply test/jdk/java/net/httpclient/ProxyServer.java]: Good point

**Review by djelinski — APPROVED** [review]

**Review by jaikiran — APPROVED** [review]
This looks OK to me.

**Pull request closed without merging**